Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,199 @@
+            {
+                EntitySystem.Get<DamageableSystem>().SetDamage(hostdamage, (hostdamage.Damage + args.DamageDelta * component.DamagePercent));
+                hostdamage.Owner.PopupMessage(Loc.GetString("guardian-entity-taking-damage"));
+            }
+        }
+
+        /// <summary>
+        /// Triggers while trying to examine an activator to see if it's used
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnActivatorExamined(EntityUid uid, GuardianCreatorComponent component, ExaminedEvent args)
+        {
+           if (component.Used == true)
+           {
+               string usedstring = Loc.GetString("guardian-activator-empty-invalid-creaton");
+               args.Message.AddMarkup("\n" + $"[color=#ba1919]{usedstring}[/color]");
+           }
+        }
+
+        /// <summary>
+        /// Adds the guardian host component to the user and spawns the guardian inside said component
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnGuardianCreated(EntityUid uid, GuardianCreatorComponent component, UseInHandEvent args)
+        {
+            //Only works if the guardian isn't already present. NOTE: this is up to preference, we can probably later allow
+            //a person to host multiple guardians, however the idea is that in a normal setting you won't get the opportunity to
+            if (!args.User.TryGetComponent<GuardianHostComponent>(out var guardianComponent))
+            {
+                if (args.User.TryGetComponent<SharedActionsComponent>(out SharedActionsComponent? action))
+                {
+                    if (component.Used == false)
+                    {
+                        var hostcomp = args.User.AddComponent<GuardianHostComponent>();
+                        var guardian = EntityManager.SpawnEntity(component.GuardianType, hostcomp.Owner.Transform.Coordinates);
+                        hostcomp.GuardianContainer.Insert(guardian);
+                        hostcomp._hostedguardian = guardian.Uid;
+                        //Ensures the guardian component on the supposed guardian entity and fills out it's synced parametres
+                        var guardiancomp = guardian.EnsureComponent<GuardianComponent>();
+                        guardiancomp.Guardian = guardian.Uid;
+                        guardiancomp.Host = args.User.Uid;
+                        //Grant the user the recall action and notify them
+                        action.Grant(ActionType.ManifestGuardian);
+                        args.User.PopupMessage(Loc.GetString("guardian-created"));
+                        //Exhaust the activator
+                        component.Used = true;
+                    }
+                    else
+                    {
```

Comment:
1. ECS, don't use IEntity methods
2. All of this functionality should go on GuardianHost startup / Initialize IMO

Reply:
Will be fixed